src/dataset/domain/models/Dataset.ts

```diff
@@ -354,6 +354,56 @@ export class Dataset {
     return false
   }
 
+  static withDatasetLabels(version: DatasetVersion): DatasetLabel[] {
+    const statusLabels = Dataset.withStatusLabel(version.publishingStatus, version.isInReview)
+    const versionLabels = Dataset.withVersionLabel(version)
+    return [...statusLabels, ...versionLabels] // combine and return
+  }
+
+  static withStatusLabel(
+    publishingStatus: DatasetPublishingStatus,
+    isInReview: boolean
+  ): DatasetLabel[] {
+    const labels: DatasetLabel[] = []
+
+    if (publishingStatus === DatasetPublishingStatus.DEACCESSIONED) {
+      labels.push(
+        new DatasetLabel(DatasetLabelSemanticMeaning.DANGER, DatasetLabelValue.DEACCESSIONED)
+      )
+    } else {
+      if (publishingStatus === DatasetPublishingStatus.DRAFT) {
+        labels.push(new DatasetLabel(DatasetLabelSemanticMeaning.DATASET, DatasetLabelValue.DRAFT))
+      }
+      const isReleased = publishingStatus === DatasetPublishingStatus.RELEASED
+      if (!isReleased) {
```

Comment:
Ok, I see what I did wrong here, thanks!  I think because my assignment if isReleased was incorrect, the "unpublished" and "deaccessioned" label were appearing at the same time, and that's why I thought I needed to change that logic.  But with the correct assignment of isReleased, I reverted to the previous logic. I think it's back to how it was now.